ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/PolygonDrawingViewModel.kt

```diff
@@ -158,20 +159,26 @@ internal constructor(
   }
 
   private fun updateVertices(newVertices: ImmutableList<Point>) {
-    mapPolygon =
-      mapPolygon.map { polygon: MapPolygon -> polygon.toBuilder().setVertices(newVertices).build() }
-    mapPolygonFlowable.onNext(mapPolygon)
+    polygonLocationOfInterest =
+      polygonLocationOfInterest.map {
+        it.copy(
+          locationOfInterest =
+            it.locationOfInterest.copy(
+              geometry = Polygon(LinearRing(newVertices.map { point -> point.coordinate }))
```

Comment:
I should probably just update the `LinearRing` constructor to take points. we keep having to write this everywhere.